### PR TITLE
Rewrite "Security recommendations" page.

### PR DESCRIPTION
**Pull request by ADubhlaoich**
### Proposed changes

Part of #5330.

This pull request rewrites and restructures the Security page in the configuration folder of the documentation.

The main body of text has been reduced to the critical information: Hugo elements and contemporary formatting has been used to emphasize important considerations for each sub-topic, and the topics have been reordered to match the most likely user journey when going through the installation process.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/content/configuration/security.md [hunk 1]
```diff
@@ -2,68 +2,50 @@
...
-The Ingress Controller requires a service account which is configured using RBAC.
-We strongly recommend using the [RBAC configuration](https://github.com/nginxinc/kubernetes-ingress/blob/v3.5.0/deployments/rbac/rbac.yaml) provided in our standard deployment configuration. It is configured with the least amount of privilege required for the Ingress Controller to work.
+NGINX Ingress Controller requires RBAC to configure a [ServiceUser](https://kubernetes.io/docs/concepts/security/service-accounts/#default-service-accounts), and provides least privilege access in its standard deployment configurations:
 
-We strongly recommend inspecting the RBAC configuration for [Manifests](https://github.com/nginxinc/kubernetes-ingress/blob/v3.5.0/deployments/rbac/rbac.yaml)
-or for [Helm](https://github.com/nginxinc/kubernetes-ingress/blob/v3.5.0/charts/nginx-ingress/templates/rbac.yaml) to understand what access the Ingress Controller service account has and to which resources. For example, by default the service account has access to all Secret resources in the cluster.
+- [Helm](https://github.com/nginxinc/kubernetes-ingress/blob/v3.5.0/deployments/rbac/rbac.yam)
```
Comment (shaun-nx) [inline docs/content/configuration/security.md:20]:
```suggestion
- [Helm](https://github.com/nginxinc/kubernetes-ingress/blob/v3.5.0/deployments/rbac/rbac.yaml)
```

Comment (vepatel) [inline docs/content/configuration/security.md:20]:
```suggestion
- [Helm](https://github.com/nginxinc/kubernetes-ingress/blob/v3.5.0/deployments/rbac/rbac.yaml)
```

File: docs/content/configuration/security.md [hunk 2]
```diff
@@ -2,68 +2,50 @@
...
+- [Manifests](https://github.com/nginxinc/kubernetes-ingress/blob/v3.5.0/deployments/rbac/rbac.yaml)
 
-### Certificates and Privacy Keys
+By default, the ServiceAccount has access to all Secret resources in the cluster.
 
-Secrets are required by the Ingress Controller for some configurations.
-[Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) are stored by Kubernetes unencrypted by default.
-We strongly recommend configuring Kubernetes to store these Secrets encrypted at rest.
-Kubernetes has [documentation](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) on how to configure this.
+## Secrets
```
[the remark this answers is not in the document]
Comment (ADubhlaoich) [reply docs/content/configuration/security.md:25]: If they can be a subheading, then what would they be under?

The previous version of this page split things into two sections unnecessarily: this iteration has each concern under its own heading.

[the remark this answers is not in the document]
Comment (vepatel) [reply docs/content/configuration/security.md:25]: Under Kubernetes, RBACs, Service Accounts and Secrets are k8s resources, while rootfilesystem, snippets and prometheus endpoint are Ingress Controller specific

**Review by danielnginx — APPROVED** [review]

**Pull request merged**